application/src/control.rs

```diff
@@ -0,0 +1,387 @@
+                    ctlpins.power_on(&self.data.config.power_rescue);
+                }
+            }
+        }
+        if let Some(action) = self.storage.take() {
+            match action {
+                StorageAction::Nop => (),
+                StorageAction::Off => {
+                    storage.power_off();
+                }
+                StorageAction::Host => {
+                    storage.connect_to_host();
+                }
+                StorageAction::DUT => {
+                    storage.connect_to_dut();
+                }
+            }
+        }
+        if let Some((pin, state)) = self.pin.take() {
+            let state = match state {
+                SetPinState::Low => PinState::Low,
+                SetPinState::High => PinState::High,
+                SetPinState::Floating => PinState::Floating,
+            };
+            match pin {
+                SetPin::Reset => {
+                    ctlpins.set_reset(state);
+                }
+                SetPin::A => {
+                    ctlpins.set_ctl_a(state);
+                }
+                SetPin::B => {
+                    ctlpins.set_ctl_b(state);
+                }
+                SetPin::C => {
+                    ctlpins.set_ctl_c(state);
+                }
+                SetPin::D => {
+                    ctlpins.set_ctl_d(state);
+                }
+            }
+        }
+    }
+}
+
+impl<B: UsbBus> UsbClass<B> for ControlClass {
+    fn get_configuration_descriptors(&self, writer: &mut DescriptorWriter) -> Result<()> {
+        writer.iad(
+            self.iface,
+            1,
+            USB_CLASS_VENDOR_SPECIFIC,
+            USB_SUBCLASS_JUMPSTARTER,
+            USB_PROTOCOL_JUMPSTARTER,
+            None,
+        )?;
+
+        writer.interface(
+            self.iface,
+            USB_CLASS_VENDOR_SPECIFIC,
+            USB_SUBCLASS_JUMPSTARTER,
+            USB_PROTOCOL_JUMPSTARTER,
+        )?;
+
+        Ok(())
+    }
+
+    fn control_in(&mut self, xfer: ControlIn<B>) {
+        let req = xfer.request();
+
+        match req {
+            &Request {
+                request_type: RequestType::Vendor,
+                recipient: Recipient::Interface,
+                index,
+                ..
+            } if index as u8 == self.iface.into() => (),
+            _ => return,
+        }
+
+        match req.request.try_into() {
+            Ok(ControlRequest::Config) => {
+                if let Ok(key) = req.value.try_into() {
+                    let cfg = &self.data.config;
+                    match key {
+                        ConfigKey::Name => {
+                            xfer.accept_with(&cfg.name).ok();
+                        }
+                        ConfigKey::Tags => {
+                            xfer.accept_with(&cfg.tags).ok();
+                        }
+                        ConfigKey::Json => {
+                            xfer.accept_with(&cfg.json).ok();
+                        }
```

Comment:
Just a note: no action is required.

We will have an issue with .json once it gets bigger than the transfer size. I don't know what the default is or what we configured. 

On the software I was starting to move name/tags/usbconsole under json , as those are details that the microcontroller itself does not care about, or use for anything,.... but the transfer size could be a good reason to stop doing such thing.